python/examples/advanced/Segments.ipynb

```diff
@@ -14,89 +14,1654 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'category': SegmentationPartition(name='category', mapper=ColumnMapperFunction(col_names=['category'], map=None, field_projector=<whylogs.core.projectors.FieldProjector object at 0x7f3a1d38e3d0>, id='31aee7544d31ada00c3bb3d94ca2e0595c42a1f21c266da65e132168914ed61fe8b1b8c99aaa1a0c5cf5e2dfbd621aa3f9700bef1f6e85f4de4ca6364f149592'), id='8ff3ae39c46814563082fd6b3a9c0cfa922a8ef8dee5e685502485ed6482e4dcf7ecc3e4f7def5451c476c5b87485c1e0d9684c7ccf0f2cf3e2ba6106ec674d8', filter=None)}"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "column_segments"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`column_segments` is a dictionary for different SegmentationPartition, with informations such as id and additional logic. For the moment, all we're interested in is that we can pass it to our `DatasetSchema` in order to generate segmented profiles while logging: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
     "import whylogs as why\n",
-    "from whylogs.api.logger.result_set import SegmentedResultSet\n",
-    "from whylogs.core.metrics.metrics import CardinalityMetric, DistributionMetric\n",
     "from whylogs.core.schema import DatasetSchema\n",
-    "from whylogs.core.segment import Segment\n",
+    "\n",
+    "results = why.log(df, schema=DatasetSchema(segments=column_segments))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since we had 11 different categories, we can expect the `results` to have 11 segments. Let's make sure that is the case:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "After profiling the result set has: 11 segments\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(f\"After profiling the result set has: {results.count} segments\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Great.\n",
+    "\n",
+    "Now, let's visualize the metrics for a single segment (the first one).\n",
+    "\n",
+    "Results can have multiple partitions, and each partition can have multiple segments. Segments within a partition are non-overlapping. Segments across partitions, however, might overlap. \n",
+    "\n",
+    "In this example, we have only one partition with 11 non-overlapping segments. Let's fatch the available segments:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "partition = results.partitions[0]\n",
+    "segments = results.segments_in_partition(partition)"
```

Comment:
you should be able to do:
`segments = results.segments()`

And you don't have to bother with the partitions.

Reply:
updated in latest commit